### PR TITLE
Create Purge site Api added php static analyser

### PR DESCRIPTION
**Pull request by Mossman1215**
Created a command and api wrapper to the purge cache endpoint in Incapsula.
Adding phpstan to the repository to catch simple mistakes in the code base which means the codebase requires php7.1 now
Phpstan caught some namespace errors which have been corrected.

### CODE REVIEW

File: src/Command/IntegrationIpsCommand.php [hunk 1]
```diff
@@ -3,9 +3,10 @@
 namespace Incapsula\Command;
 
 use Symfony\Component\Console\Helper\Table;
-use Symfony\Component\Console\Input\Inputinterface;
+use Symfony\Component\Console\Input\InputInterface;
 use Symfony\Component\Console\Input\InputOption;
 use Symfony\Component\Console\Output\OutputInterface;
+use Incapsula\Api;
```
Comment (halkyon) [inline src/Command/IntegrationIpsCommand.php]: this doesn't appear to be used, there's no references to the Api class from that namespace in the file... not sure why phpstan would have picked this up?

File: src/Api/SitesApi.php [hunk 2]
```diff
@@ -69,4 +69,21 @@ public function removeCustomCertificate($siteId)
             'site_id' => $siteId,
         ]);
     }
+    /**
```
Comment (halkyon) [inline src/Api/SitesApi.php]: could you put newlines around each function/docblock block? the formatter fixer doesn't seem to pick up on this unfortunately

File: src/Api/SitesApi.php [hunk 3]
```diff
@@ -69,4 +69,21 @@ public function removeCustomCertificate($siteId)
...
+     * See incapsula docs for details
+     * https://docs.incapsula.com/Content/API/sites-api.htm#Purge
+     * @return array
+     */
+    public function purgeCache($siteId, $purgePattern = "")
+    {
+        return $this->client->send(sprintf('%s/cache/purge', $this->apiUri), [
+            'site_id' => $siteId,
+            'purge_pattern' => $purgePattern
+            ]);
```
Comment (halkyon) [inline src/Api/SitesApi.php]: one indent back please!

File: src/Command/PurgeSiteCommand.php [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    protected $resourcePattern;
+    protected function configure()
+    {
+        parent::configure();
+
+        $this
+            ->setName('site:purge')
+            ->setDescription('Purge URLs from a given site id')
+            ->addArgument('site-id',InputArgument::REQUIRED,'incapsula id of site to purge')
+            ->addArgument('resource-pattern',InputArgument::OPTIONAL,'string to match in the URL to be purged',"")
```
Comment (halkyon) [inline src/Command/PurgeSiteCommand.php]: add space after the `,` on the site-id and resource-pattern arguments

File: src/Command/PurgeSiteCommand.php [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    }
+    /**
+     * @param InputInterface  $input
+     * @param OutputInterface $output
+     *
+     * @return int
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->client->sites()->purgeCache($this->siteId,$this->resourcePattern);
```
Comment (halkyon) [inline src/Command/PurgeSiteCommand.php]: same here: space after `,`

**Comment by halkyon** [comment]
looks like circleci didn't run, have turned on circleci build forks/pull requests so it should run next time you push up

**Comment by Mossman1215** [comment]
Ah I don't have php cs fixer installed on my home laptop :P

**Comment by Mossman1215** [comment]
how about that eh? finally got my local environment setup and I learned about php-cs-fixer!

**Pull request merged**